lucet-idl/src/data_description_helper.rs

```diff
@@ -0,0 +1,171 @@
+use super::cache::*;
+use super::errors::*;
+use super::generators::*;
+use super::pretty_writer::*;
+use lucet_idl::validate::*;
+use std::io::prelude::*;
+
+/// A convenient structure holding a data type, its name and
+/// its internal IDL representation
+#[derive(Debug, Clone)]
+pub struct DataTypeEntry<'t> {
+    pub id: DataTypeId,
+    pub name: &'t Name,
+    pub data_type: &'t DataType,
+}
+
+/// Transforms a `DataDescription`
+/// We definitely need a better name for it
+#[derive(Debug, Clone)]
+pub struct DataDescriptionHelper {
+    pub data_description: DataDescription,
+}
+
+impl DataDescriptionHelper {
+    /// Retrieve information about a data type given its identifier
+    pub fn get(&self, id: DataTypeId) -> DataTypeEntry<'_> {
```

Comment:
Could this move to be on `impl DataDescription` itself? The remainder would then be just functions writing out for whichever backend, making for a name like `DataDescriptionWriter` or something along those lines.

Reply:
`DataDescription` only belongs to the lexer. It isn't supposed to know anything about code generation.

Unfortunately, that structure is not very convenient to work with for code generation.

Granted, `DataDescriptionHelper` is a bad and confusing name. I'm not good at naming things, so suggestions are welcome :)